src/Window.cpp

```diff
@@ -1299,40 +1513,68 @@ void Window::updateMouseProgress(double progress) {
     }
     double fraction = 1.0 - (remaining / required);
 
-    // Calculate the current translation and rotation
-    Coordinate startingTranslation =
-        getCenterOfTile(m_startingLocation.first, m_startingLocation.second);
-    Coordinate destinationTranslation =
-        getCenterOfTile(destinationLocation.first, destinationLocation.second);
-    Angle startingRotation =
-        DIRECTION_TO_ANGLE().value(m_startingDirection);
-    Coordinate currentTranslation =
-        startingTranslation * (1.0 - fraction) +
-        destinationTranslation * fraction;
-    Angle currentRotation =
-        startingRotation * (1.0 - fraction) +
-        destinationRotation * fraction;
-
-    // Teleport the mouse, reset movement state if done
+    // Calculate the current translation and rotation for moving straight or turn 45 or 90 degree
+    Coordinate startingTranslation = getCenterOfTile(m_startingLocation.first, m_startingLocation.second);
+    Coordinate destinationTranslation = getCenterOfTile(destinationLocation.first, destinationLocation.second);
+
+    Angle startingRotation = DIRECTION_TO_ANGLE().value(m_startingDirection);
+    Coordinate currentTranslation = startingTranslation * (1.0 - fraction) + destinationTranslation * fraction;
+    Angle currentRotation =         startingRotation * (1.0 - fraction) + destinationRotation * fraction;
+
+    // Calculate the current translation and rotation for a curv turn
+    if (m_movement == Movement::RUN_RIGHT || m_movement == Movement::RUN_LEFT) {
```

Comment:
The idea here is to move along an arc, correct? If so, perhaps we should extract this logic into its own method to make it easier to read.